indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1137,6 +1394,20 @@ public void gracefulShutdownInternal() throws ExecutionException, InterruptedExc
   @VisibleForTesting
   public void resetInternal(DataSourceMetadata dataSourceMetadata)
   {
+    // clear queue for kafka lags
+    if (enableDynamicAllocationTasks && queue != null) {
+      try {
+        lock.lock();
+        queue.clear();
+      }
+      catch (Exception e) {
+        log.warn(e, "Error,when clear queue in rest action");
```

Comment:
If this clear action is failed. The collection of lag points may be not as accurate as expected. May interfere with scale action. But this failing rarely happens and the  impact is acceptable. Because scale action is graceful and no data will lose or duplicate.